frontend/gateway/grpcclient/client.go

```diff
@@ -28,18 +29,41 @@ func Current() (client.Client, error) {
 
 	c := pb.NewLLBBridgeClient(conn)
 
-	_, err = c.Ping(ctx, &pb.PingRequest{})
+	resp, err := c.Ping(ctx, &pb.PingRequest{})
 	if err != nil {
 		return nil, err
 	}
 
-	return &grpcClient{client: c, opts: opts(), sessionID: sessionID()}, nil
+	if resp.FrontendAPICaps == nil {
+		resp.FrontendAPICaps = defaultCaps()
+	}
+
+	return &grpcClient{
+		client:    c,
+		opts:      opts(),
+		sessionID: sessionID(),
+		workers:   workers(),
+		caps:      pb.Caps.CapSet(resp.FrontendAPICaps),
+	}, nil
+}
+
+// defaultCaps returns the capabilities that were implemented when capabilities
+// support was added
+func defaultCaps() []apicaps.PBCap {
```

Comment:
@tonistiigi I would unit test this function to make sure its output never changes.